### ISSUE TITLE
Ambiguous record warning thrown while using implicit join syntax on tables which have a common column name

### ISSUE THREAD

**Issue by rajki** [jOOQ/jOOQ#7877]
### Expected behavior and actual behavior: 
Fetching a column from the parent table record using the implicit join syntax should work as intended (fetch the correct value). However, if the parent table and child table share a column name, a SQL warning about "ambiguous names" is being thrown and the value of the child record column is fetched for the parent record column.

### Steps to reproduce the problem (include example code if possible):
Schema: 3 tables named Account, Product, Organisation. Account table has a foreign key to product table's ID column as ACCOUNT.PRODUCT_ID and to organisation table's ID column as ACCOUNT.ORGANISATION_ID. All 3 tables have a field called UUID.

Example query: 

```java
Record4<String, String, String, String> accountInformation = 
    dsl.select(account.UUID, account.DOMAIN, account.organisation().UUID, account.product().UUID)
                .from(account).where(account.UUID.eq(uuid))
                .fetchOne();
        if(accountInformation == null) {
            // throw exception
        }
        // fetch information
        accountInformation.get(account.UUID),
        accountInformation.get(account.DOMAIN),
        accountInformation.get(product.UUID)
        accountInformation.get(organisation.UUID)
```

The last two get statements return account's UUID value instead of product and organisation's UUID value.

### Versions:

- jOOQ: 3.11
- Java: 1.8
- Database (include vendor): H2
- OS: Mac OSX
- JDBC Driver (include name if inofficial driver): h2-database

**Comment by lukaseder** [comment]
Thanks a lot for your report.

The problem here is the fact that you're trying to fetch a column that doesn't really exist, *technically*, even if it seems to be correctly qualified. The currently correct way to get a hold of the three `UUID` values is this:

```java
accountInformation.get(account.UUID),
accountInformation.get(account.organisation().UUID)
accountInformation.get(account.product().UUID)
```

If you don't use the exact expression that you used in the query also to get the value from the record, then jOOQ will find the first matching column by unqualified name and report an ambiguity in the logs.

I do think that in your particular case, we should have an additional step where partially qualified names will still match with higher priority than unqualified names. I.e. the fact that you used the specific table of the `UUID` for qualification should be relevant, and non ambiguous.

**Comment by rajki** [comment]
Thanks a lot for the quick turnaround on this, @lukaseder!

**Comment by lukaseder** [comment]
I think this has been solved with #14671. I can no longer reproduce any warning. Closing this as a duplicate of #14671
